sympy/combinatorics/fp_groups.py

```diff
@@ -688,4 +798,410 @@ def coset_enumeration_c(fp_grp, Y):
+    >>> C221.table.append([None]*4)
+    >>> C221.p = [0, 1, 2]
+    >>> C221.table[2][2] = 0; C221.table[0][3] = 2;
+    >>> C221.table
+    [[1, 1, 1, 2], [0, 0, None, 0], [None, None, 0, None]]
+    >>> C221.scan(1, y**3)
+    >>> C221.table
+    >>> C221.scan(2, y**3)
+    [[1, 1, 1, 2], [0, 0, 2, 0], [None, None, 0, 1]]
+    >>> first_in_class(C221)
+    True
+    >>> C221.table
+    [[1, 1, 1, 2], [0, 0, 2, 0], [None, None, 0, 1]]
+    >>> new_C221 = C221.copy()
+    >>> new_C221.table[2][0] = 2; new_C221.table[2][1] = 2
+    >>> first_in_class(new_C221)
+    False
+    >>> C2211 = C221.copy()
+    >>> C2211.table.append([None]*4)
+    >>> C2211.p = [0, 1, 2, 3]
+    >>> C2211.table[2][0] = 3; C2211.table[2][1] = 3
+    >>> C2211.table
+    [[1, 1, 1, 2], [0, 0, 2, 0], [3, 3, 0, 1], [None, None, None, None]]
+    >>> C2211.table[3][0] = 2; C2211.table[3][1] = 2;
+    >>> C2211.table
+    [[1, 1, 1, 2], [0, 0, 2, 0], [3, 3, 0, 1], [2, 2, None, None]]
+    >>> C2211.table[3][2] = 3; C2211.table[3][3] = 3
+    >>> C2211.table
+    [[1, 1, 1, 2], [0, 0, 2, 0], [3, 3, 0, 1], [2, 2, 3, 3]]
+    >>> first_in_class(C2211)
+    False
+    >>> C23 = C2.copy()
+    >>> C23.table
+    [[1, 1, 2, None], [0, 0, None, None]]
+    >>> C23.table.append([None]*4);
+    >>> C23.p = [0, 1, 2]
+    >>> C23.table[0][2] = 2; C23.table[2][3] = 0
+    >>> C23.table
+    [[1, 1, 2, None], [0, 0, None, None], [None, None, None, 0]]
+    >>> C231 = C23.copy()
+    >>> C231.table[0][3] = 1; C231.table[1][2] = 0
+    >>> C231.table
+    [[1, 1, 2, 1], [0, 0, 0, None], [None, None, None, 0]]
+    >>> first_in_class(C231)
+    False
+    >>> C232 = C23.copy()
+    >>> C232.table
+    [[1, 1, 2, None], [0, 0, None, None], [None, None, None, 0]]
+    >>> C232.p
+    [0, 1, 2]
+    >>> C232.table.append([None]*4)
+    >>> C232.table
+    [[1, 1, 2, None],
+     [0, 0, None, None],
+     [None, None, None, 0],
+     [None, None, None, None]]
+    >>> C232.table[0][3] = 3; C232.table[3][2] = 0
+    >>> C232.table
+    [[1, 1, 2, 3],
+     [0, 0, None, None],
+     [None, None, None, 0],
+     [None, None, 0, None]]
+    >>> C232.scan(2, y**3)
+    >>> C232.table
+    [[1, 1, 2, 3], [0, 0, None, None], [None, None, 3, 0], [None, None, 0, 2]]
+    >>> C232.table[1][2] = 1; C232.table[1][3] = 1
+    >>> C232.table
+    [[1, 1, 2, 3], [0, 0, 1, 1], [None, None, 3, 0], [None, None, 0, 2]]
+    >>> first_in_class(C232)
+    False
+
+    """
+    n = C.n
+    lamda = -1
+    nu = [None]*n
+    mu = []
+    breaker = False
+    for alpha in range(1, n):
+        # reset ν to "None" after previous value of α
+        for beta in range(lamda+1):
+            nu[mu[beta]] = None
+        # try α as the new point 0 in Ω_C_α
+        try:
+            mu[0] = alpha
+        except IndexError:
+            mu.append(alpha)
+        nu[alpha] = 0
+        # compare corresponding entries in C and C_α
+        lamda = 0
+        for beta in range(C.n):
+            if breaker:
```

Comment:
Perhaps this could be after the following loop. And probably `breaker` should be reset. (Something like `next_alpha` might be more descriptive from a mathematicians point of view.)